### PR TITLE
Bug 454: when clicking on message in inbox the state of the unread me…

### LINKED ISSUE #454: Bug: Message badge not updating after message read
### Bug Title: Bug: Message badge not updating after message read

---
### Bug Description:

Related to issue #427 and PR #437

At the moment, the badge value for the message remains unchanged even after a message been read.

---
### Reproduction Steps:    
**1.**    Read an unread message
**2.**   See bug

---
### Expected Behavior:

The badge value should decrease each time a message been read. The badge should vanish after all message been read.

---
### Screenshots (If applicable):

https://github.com/codeforpdx/PASS/assets/14917816/b40f199c-e06b-425d-9599-c28ce3ba4b47

---
### Possible Solution (optional):

Have the badge value updated upon refresh or when a message have been read. At the very least, it'll be 1 less than the initial value shown.

### PR DESCRIPTION
**Pull request by bingeboy**
# Message Badge Not Updating After Message Read

## This PR:
Resolves #454 
Added the React update to the `<Messages />` page so the state will stay in sync with the podserver message read status for the inbox. 

## Video:
[screen-recorder-mon-oct-23-2023-12-58-19.webm](https://github.com/codeforpdx/PASS/assets/96048/5407c9e0-9234-435c-8831-38f64fb5834c)

### How to test

1. Login to podserver and navigate to the messages section
2. send more than one message to a pod you have access to or yourself
3. check that the count is on the message icon on the top right
4. state should update once the podserver is updated

## Issues needing discussion/feedback (optional):
**1.**  Need to add test coverage... could be an additional issue

### CODE REVIEW

**Comment by leekahung** [comment]
Think we can change this PR into a draft for the time being if this is a WIP

**Comment by leekahung** [comment]
Hey @bingeboy. Just wanted to check up on the progress of this draft. Are you making any headways? Let me know if you need anything, I'll see what I can help out on.

**Comment by bingeboy** [comment]
@leekahung sry haven't had time to work since we spoke. I'm focusing on the issue today. I have the state updating based on the inbox unread value. I'm playing around with it more now and going to add a test. Feel free to pull the branch and try it out.

**Comment by bingeboy** [comment]
[screen-recorder-mon-oct-23-2023-12-58-19.webm](https://github.com/codeforpdx/PASS/assets/96048/5407c9e0-9234-435c-8831-38f64fb5834c)

@leekahung check this video. Sort of lags at times but that's seems to be when it's waiting for a response from the podserver

**Comment by leekahung** [comment]
> @leekahung check this video. Sort of lags at times but that's seems to be when it's waiting for a response from the podserver

Yes, the lagging is due to the message being created to Solid (the message content itself and the Solid Thing) and send on Solid (to the recipient and a copy to your Outbox). How it's built now was not intended for rapid sending of message to one person. Simply one message at a time and waiting for a reply.

File: src/components/Messages/MessagePreview.jsx [hunk 1]
```diff
@@ -74,7 +74,10 @@ const MessagePreview = ({ message, folderType }) => {
               sx={{ padding: '10px' }}
             >
               {messageInfo.map((info, index) => (
-                <Grid item xs={info.xs_value} sx={{ opacity: message.readStatus ? '0.5' : '1' }}
+                <Grid
+                  item
+                  xs={info.xs_value}
+                  sx={{ opacity: message.readStatus ? '0.5' : '1' }}
                   key={info + String(index)}
```
Comment (bingeboy) [inline src/components/Messages/MessagePreview.jsx]: FYI this key was updated since replies to messages were running into key errors with title.

Comment (leekahung) [reply src/components/Messages/MessagePreview.jsx]: No problem, any bug fix along the way is more than welcomed. Feel free to use anything from that message object to use as the unique key or to create a unique key to be used.

**Comment by leekahung** [comment]
Just pulled the branch up a short while ago. It looks like the feature to update the badge state is working great from what I saw.

Let me know if you think it'll be ready to review and change from WIP and draft.

Great work!

**Comment by bingeboy** [comment]
Updated for peer review.

**Review by timbot1789 — CHANGES_REQUESTED** [review]
Still need to review, but need 1 unit test to show the change

**Review by timbot1789 — COMMENTED** [review]
Code itself looks good. Just waiting on unit test. Nice work!

**Comment by bingeboy** [comment]
Currently the test added checks that `<Messages />` has rendered. The messages are not loading in the test so I'm trying to sort that out next.

**Comment by bingeboy** [comment]
Getting back at this today... had a few days dealing with some heavy machines being moved around my property. @timbot1789 no messages are loading on the page in the test. I believe that's related to the the mocked message object.

**Comment by bingeboy** [comment]
@timbot1789 hey @leekahung and I fixed all merge conflicts.
 I haven't had a chance to add more tests but should have them tomorrow or else will open another ticket if I run into any more issues as we discussed.

**Comment by leekahung** [comment]
> @timbot1789 hey @leekahung and I fixed all merge conflicts. I haven't had a chance to add more tests but should have them tomorrow or else will open another ticket if I run into any more issues as we discussed.

Yeah, I'm good with leaving the unit test for the message badge in a follow-up PR if needed.

**Review by timbot1789 — APPROVED** [review]

**Comment by bingeboy** [comment]
Since new tests will need review I'm going to squash & merge now

**Pull request merged**